pandas/io/excel.py

```diff
@@ -465,7 +465,7 @@ def _parse_cell(cell_contents, cell_typ):
                 if not squeeze or isinstance(output[asheetname], DataFrame):
                     output[asheetname].columns = output[
                         asheetname].columns.set_names(header_names)
-            except StopIteration:
+            except EmptyDataError:
```

Comment:
No I am not.  I am specifically raising `EmptyDataError` in specific locations where an empty `StopIteration` used to be raised.  I am not gloming them together.  The `try-except` will let generic `ValueError` bubble up still.